services/DemographicsService.js

```diff
@@ -6,35 +6,74 @@
 class DemographicsService {
 
     /**
-     * Constructs a new service backed by the provided
-     * `dbSession` database session
-     * @param dbSession a connected database session
+     * Constructs a new service backed by the AWS docClient
      */
     constructor(dbSession) {
-        this.session = dbSession;
+        var AWS = require("aws-sdk");
+
+        AWS.config.update({
+          region: "us-east-1",
+          endpoint: "https://dynamodb.us-east-1.amazonaws.com"
+        });
+        //AWS.config.setPromisesDependency(null);
+        this.docClient = new AWS.DynamoDB.DocumentClient();
     }
 
     /**
      * Returns demographic details for the provided set of pubmed IDs.
      * @param ids an iterable of pubmed IDs
      * @return an iterable of `Promise`s of demographic details
      */
-    getDemographicDetailsForIds(ids) {
-
-        let demoQueryForId = (id) => {
-
-            // TODO: replace with dynamoDB query
+    getDemographicDetailsForIds(pmcid) {
+        let demoQueryForId = (pmcid) => {
             return new Promise((resolve, reject) => {
-                resolve({
-                    pmid: id,
-                    malePercent: 50,
-                    femalePercent: 50
-                });
+                this.docClient.get(makeParamsFromPmcid(pmcid), function(err, data){
```

Comment:
I think it would be _highly_ desirable to use `batchGet` here. Otherwise we are doing 20+ network roundtrips when we could be doing just one